src/validators/user-info-request-validator.ts

```diff
@@ -0,0 +1,68 @@
+import { UserInfoRequestError } from "../errors/user-info-request-error";
+import { IncomingHttpHeaders } from "http";
+import { signedJwtValidator } from "./signed-jwt-validator";
+import { Config } from "../config";
+import { logger } from "../logger";
+
+const AuthorisationHeaderKey: string = "authorization";
+const ValidScopes: string[] = ["openid", "email", "phone"];
+
+export const userInfoRequestValidator = async (
+  userInfoRequestHeaders: IncomingHttpHeaders
+): Promise<
+  | { valid: true }
+  | {
+      valid: false;
+      error: UserInfoRequestError;
+    }
+> => {
+  for (const key of Object.keys(userInfoRequestHeaders)) {
+    userInfoRequestHeaders[key.toLowerCase()] = userInfoRequestHeaders[key];
+  }
+  const accessToken = userInfoRequestHeaders[AuthorisationHeaderKey];
+  let match;
+  try {
+    match = /^Bearer (?<token>.*)$/.exec(accessToken as string);
+  } catch (error) {
+    logger.error("Failed To parse authorisation header.", error);
+    return { valid: false, error: UserInfoRequestError.MISSING_TOKEN };
+  }
+
+  if (!match?.groups?.token) {
+    logger.warn("Missing access token in authorisation header.");
+    return { valid: false, error: UserInfoRequestError.MISSING_TOKEN };
+  }
+
+  const jwtResult = await signedJwtValidator(match.groups.token);
+  if (!jwtResult.valid) {
+    logger.warn("Failed to verify jwt signature.");
+    return { valid: false, error: UserInfoRequestError.INVALID_TOKEN };
+  }
+
+  const config = Config.getInstance();
+  const config_client_id = config.getClientId();
+  const config_scopes = config.getScopes();
+
+  const { client_id, scope } = jwtResult.payload;
+
```

Comment:
Also client id null check

Reply:
Hmm, I assumed it would be fine to consolidate this into one check as we should throw an invalid token error if it's null or doesn't match. It does look like right now we would allow a null value in the config though, so I guess you're right:
```
if (clientConfiguration.clientId !== undefined) {
        config.setClientId(clientConfiguration.clientId);
}
```

Reply:
Yeah, I think it just makes it a bit more clear!